src/lib/es6.d.ts

```diff
@@ -127,7 +127,7 @@ interface ObjectConstructor {
       * @param target The target object to copy to.
       * @param source The source object from which to copy properties.
       */
-    assign<T, U>(target: T, source: U): T & U;
+    assign<T, U>(target: T, source: U | null | undefined): T & U;

```

Comment:
`T & null` and `T & undefined` should not be a problem, because `null` and `undefined` add no new member to the type? In fact, I think the type system could (should?) simplify `T & null & undefined` to just `T`.

If it were `T | null` that would be a real problem.